henson-plugin/build.gradle

```diff
@@ -0,0 +1,65 @@
+apply plugin: 'groovy'
+apply plugin: 'java-gradle-plugin'
+
+
+sourceSets {
+    functionalTest {
+        groovy.srcDir file('src/functTest/groovy')
+        resources.srcDir file('src/functTest/resources')
+        compileClasspath += sourceSets.main.output + configurations.testRuntime
+        runtimeClasspath += output + compileClasspath
+    }
+}
+
+gradlePlugin {
+    plugins {
+        hensonPlugin {
+            id = 'dart.henson-plugin'
+            implementationClass = 'dart.henson.plugin.HensonPlugin'
+        }
+    }
+    testSourceSets sourceSets.functionalTest
+}
+
+task functionalTest(type: Test) {
+    description = 'Runs the functional tests.'
+    group = 'verification'
+    testClassesDirs = sourceSets.functionalTest.output.classesDirs
+    classpath = sourceSets.functionalTest.runtimeClasspath
+    mustRunAfter test
+}
+
+check.dependsOn functionalTest
+
+project.afterEvaluate {
+    tasks.getByName('compileGroovy').doFirst {
+        //we create a file in the plugin project containing the current project version
+        //it will be used later by the plugin to add the proper version of
+        //dependencies to the project that uses the plugin
+        def prop = new Properties()
+        def propFile = new File("${project.rootDir}/henson-plugin/src/main/resources/build.properties")
```

Comment:
Does the D&H version go here? If so, would be nice to use it for the composite builds needed for D&H sample.

Reply:
Yes, if there is no better way to do it.